File: src/main/java/bunny/mybatis/FieldType.java
package bunny.mybatis;

import java.util.HashMap;

public class FieldType {

    private static final HashMap<String, String> MAP = new HashMap<>();

    static {
        MAP.put("String", "'");
        MAP.put("Boolean", "");
        MAP.put("Character", "'");
        MAP.put("Byte", "");
        MAP.put("Short", "");
        MAP.put("Integer", "");
        MAP.put("Long", "");
        MAP.put("Float", "");
        MAP.put("Double", "");
        MAP.put("BigDecimal", "");
        MAP.put("Date", "'");
        MAP.put("Timestamp", "'");
        MAP.put("LocalDate", "'");
        MAP.put("LocalTime", "'");
        MAP.put("LocalDateTime", "'");
    }

    public static boolean isNormalType(String typeName) {
        return MAP.containsKey(typeName);
    }

    public static String getDeepStartType(String typeName) {
        if (typeName == null || typeName.isEmpty()) {
            return typeName;
        }

        int i = typeName.indexOf("<");
        return i < 0 ? typeName : typeName.substring(0, i);
    }

    public static String getDefaultValue(String key) {
        return MAP.get(key) == null ? MAP.get("String") : MAP.get(key);
    }
}

File: src/main/kotlin/bunny/plugins/mybatis/FieldType.kt
package bunny.plugins.mybatis

object FieldType {
    private val MAP = HashMap<String, String?>()

    init {
        MAP["String"] = "'"
        MAP["Boolean"] = ""
        MAP["Character"] = "'"
        MAP["Byte"] = ""
        MAP["Short"] = ""
        MAP["Integer"] = ""
        MAP["Long"] = ""
        MAP["Float"] = ""
        MAP["Double"] = ""
        MAP["BigDecimal"] = ""
        MAP["Date"] = "'"
        MAP["Timestamp"] = "'"
        MAP["LocalDate"] = "'"
        MAP["LocalTime"] = "'"
        MAP["LocalDateTime"] = "'"
    }

    @JvmStatic
    fun isNormalType(typeName: String): Boolean {
        return MAP.containsKey(typeName)
    }

    fun getDeepStartType(typeName: String?): String? {
        if (typeName == null || typeName.isEmpty()) {
            return typeName
        }

        val i = typeName.indexOf("<")
        return if (i < 0) typeName else typeName.substring(0, i)
    }

    @JvmStatic
    fun getDefaultValue(key: String): String? {
        return if (MAP[key] == null) MAP["String"] else MAP[key]
    }
}
